api-js/src/user/mutations/update-user-profile.js

```diff
@@ -77,6 +77,17 @@ export const updateUserProfile = new mutationWithClientMutationId({
       throw new Error(i18n._(t`Unable to update profile. Please try again.`))
     }
 
+    // Check to see if user name is already in use
+    if (userName !== '') {
+      const checkUser = await userLoaderByUserName.load(userName)
+      if (typeof checkUser !== 'undefined') {
+        console.warn(
+          `User: ${userKey} attempted to update their username, but the username is already in use.`,
+        )
+        throw new Error(i18n._(t`Unable to update profile. Please try again.`))
```

Comment:
We might need to be more specific than this -- a user might think this is a system error and not something they need to address.